roxie/ccd/ccdserver.cpp

```diff
@@ -27239,10 +27239,43 @@ class CActivityGraph : implements IActivityGraph, implements IThorChildGraph, im
         CActivityGraph * container;
     } graphCodeContext;
 
+    class SinkThread : public CThreadedPersistent, implements IThreaded
```

Comment:
CThreadedPersistent shoulod be a member of this class rather than a base class (its functions are never called from the outside).

Also, CThreadedPersistent is currently derived from CInterface, but there is no need for it to be - I suspect it is historical - that should be removed as a base class.

Reply:
Although because you are adding them to an array SinkThread should be derived from CInterface.